src/openzaak/utils/validators.py

```diff
@@ -276,14 +276,48 @@ def __call__(self, url: str):
 
 class UniekeIdentificatieValidator(_UniekeIdentificatieValidator):
 
-    partial_message = _(
-        "Deze identificatie ({identificatie}) bestaat al binnen de organisatie"
-    )
-
-    @property
-    def message(self):
-        return self.partial_message.format(identificatie=self.identificatie)
+    message = _("Deze identificatie ({identificatie}) bestaat al binnen de organisatie")
 
     def __call__(self, attrs: dict, serializer):
-        self.identificatie = attrs.get(self.identificatie_field)
-        return super().__call__(attrs, serializer)
+        instance = getattr(serializer, "instance", None)
+        identificatie = attrs.get(self.identificatie_field)
+        if not identificatie:
+            if instance:
+                # In case of a partial update
+                identificatie = instance.identificatie
+            else:
+                # identification is being generated, and the generation checks for
+                # uniqueness
+                return
+
+        organisatie = attrs.get(self.organisatie_field)
+        pk = instance.pk if instance else None
+
+        # if we're updating an instance, setting the current values will not
+        # trigger an error because the instance-to-be-updated is excluded from
+        # the queryset. If either bronorganisatie or identificatie changes,
+        # and it already exists, it will raise a validation error
+        combination_exists = (
+            serializer.Meta.model.objects
+            # in case of an update, exclude the current object. for a create, this
+            # will be None
+            .exclude(pk=pk)
+            .filter(
+                **{
+                    self.organisatie_field: organisatie,
+                    self.identificatie_field: identificatie,
+                }
+            )
+            .exists()
+        )
+
+        # only part changed
+        if combination_exists:
+            raise serializers.ValidationError(
+                {
+                    self.identificatie_field: self.message.format(
+                        identificatie=identificatie
```

Comment:
Am I correct that the only change of the validator is this line of code? 
In this case please move this change to the `commonground-api-common` like Steven suggested